src/com/google/enterprise/adaptor/prebuilt/RegexDecisionFilter.java

```diff
@@ -0,0 +1,285 @@
+ * restrict the search to only {@code Metadata} or {@code params}, respectively.
+ * Most keys/values of interest will normally be specified in the document's
+ * {@code Metadata}, but some key/values of interest (e.g. ContentType, DocId)
+ * exist in the document's {@code params}.
+ *
+ * <p>Example: skip documents that have a {@code NoIndex} metadata key or params
+ * key, regardless of value:
+ * <pre><code>
+   metadata.transform.pipeline=regexDecisionFilter
+   metadata.transform.pipeline.regexDecisionFilter.factoryMethod=com.google.enterprise.adaptor.prebuilt.SkipDocumentFilter.create
+   metadata.transform.pipeline.regexDecisionFilter.key=NoIndex
+   metadata.transform.pipeline.regexDecisionFilter.decision=do-not-index
+   </code></pre>
+ *
+ * <p>Example: drop the content of documents that have a {@code ContentLength}
+ * greater than or equal to 100 megabytes:
+ * <pre><code>
+   metadata.transform.pipeline=regexDecisionFilter
+   metadata.transform.pipeline.regexDecisionFilter.factoryMethod=com.google.enterprise.adaptor.prebuilt.RegexDecisionFilter.create
+   metadata.transform.pipeline.regexDecisionFilter.key=ContentLength
+   metadata.transform.pipeline.regexDecisionFilter.pattern=0*[1-9][0-9]{8,}
+   metadata.transform.pipeline.regexDecisionFilter.decision=do-not-index-content
+   </code></pre>
+ *
+ * <p>Example: skips documents whose Metadata {@code Classification} property
+ * is neither {@code PUBLIC} nor {@code DECLASSIFIED}:
+ * <pre><code>
+   metadata.transform.pipeline=regexDecisionFilter
+   metadata.transform.pipeline.regexDecisionFilter.factoryMethod=com.google.enterprise.adaptor.prebuilt.RegexDecisionFilter.create
+   metadata.transform.pipeline.regexDecisionFilter.key=Classification
+   metadata.transform.pipeline.regexDecisionFilter.pattern=(PUBLIC)|(DECLASSIFIED)
+   metadata.transform.pipeline.regexDecisionFilter.decideOnMatch=false
+   metadata.transform.pipeline.regexDecisionFilter.decision=do-not-index
+   metadata.transform.pipeline.regexDecisionFilter.corpora=metadata
+   </code></pre>
+ */
+public class RegexDecisionFilter implements MetadataTransform {
+  /**
+   * Which collections of keys/values to search.  Metadata, params, or both.
+   */
+  private enum Corpora {
+    METADATA("metadata"),
+    PARAMS("params"),
+    METADATA_OR_PARAMS("metadata or params");
+
+    private final String name;
+
+    private Corpora(String n) {
+      name = n;
+    }
+
+    public static Corpora from(String val) {
+      if ("metadata".equalsIgnoreCase(val)) {
+        return Corpora.METADATA;
+      }
+      if ("params".equalsIgnoreCase(val)) {
+        return Corpora.PARAMS;
+      }
+      return METADATA_OR_PARAMS;
+    }
+
+    public String toString() {
+      return name;
+    }
+  };
+
+  private static final Logger log
+      = Logger.getLogger(RegexDecisionFilter.class.getName());
+
+  /** The name of the key (either Metadata key or params key) to match. */
+  private String key;
```

Comment:
This rename from propertyName is for consistency with the PrebuiltTransforms? If we're doing that, can we rename corpora to something better?